CONTRIBUTING.md

```diff
@@ -28,3 +28,46 @@ To check formatting and code quality, run `swift run BuildTool lint`. Run with `
 - We should aim to make it easy for consumers of the SDK to be able to mock out the SDK in the tests for their own code. A couple of things that will aid with this:
   - Describe the SDK’s functionality via protocols (when doing so would still be sufficiently idiomatic to Swift).
   - When defining a `struct` that is emitted by the public API of the library, make sure to define a public memberwise initializer so that users can create one to be emitted by their mocks. (There is no way to make Swift’s autogenerated memberwise initializer public, so you will need to write one yourself. In Xcode, you can do this by clicking at the start of the type declaration and doing Editor → Refactor → Generate Memberwise Initializer.)
+- When writing code that implements behaviour specified by the Chat SDK features spec, add a comment that references the identifier of the relevant spec item.
+
+### Testing guidelines
+
+#### Attributing tests to a spec point
+
+When writing a test that relates to a spec point from the Chat SDK features spec, add a comment that contains one of the following tags:
+
+- `@spec <spec-item-id>` — The test case directly tests all the functionality documented in the spec item.
+- `@specOneOf(m/n) <spec-item-id>` — The test case is the m<sup>th</sup> of n test cases which, together, test all the functionality documented in the spec item.
+- `@specPartial <spec-item-id>` — The test case tests some, but not all, of the functionality documented in the spec item. This is different to `@specOneOf` in that it implies that the test suite does not fully test this spec item.
+
+The `<spec-item-id>` parameter should be a spec item identifier such as `CHA-RL3g`.
+
+Each of the above tags can optionally be followed by a hyphen and an explanation of how the test relates to the given spec item.
+
+Examples:
+
+```swift
+// @spec CHA-EX3f
+func test1 { … }
+```
+
+```swift
+// @specOneOf(1/2) CHA-EX2h — Tests the case where the room is FAILED
+func test2 { … }
+
+// @specOneOf(2/2) CHA-EX2h — Tests the case where the room is SUSPENDED
+func test3 { … }
+```
+
+```swift
+// @specPartial CHA-EX1h4 - Tests that we retry, but not the retry attempt limit because we’ve not implemented it yet
+func test4 { … }
+```
+
+In [#46](https://github.com/ably-labs/ably-chat-swift/issues/46), we’ll write a script that uses these tags to generate a report about how much of the feature spec we’ve implemented.
+
+#### Marking a spec point as untested
+
+In addition to the above, you can add the following as a comment anywhere in the test suite:
+
+- `@specUntested <spec-item-id> - <explanation>` — This indicates that the SDK implements the given spec point, but that there are no automated tests for it. This should be used sparingly; only use it when there is no way to test a spec point. It must be accompanied by an explanation of why this spec point is not tested.
```

Comment:
> detect tests without any of those `spec...` marks

For #46, I wasn't planning on "detecting" tests, which would require more code parsing than I’d like to spend time on right now. (Even JS, which [has a `@nospec` tag](https://github.com/ably/ably-js/blob/8c9ce8b0ac087abb8861891b61372863d69b9dc0/CONTRIBUTING.md?plain=1#L65), doesn't really do much with this tag; there's nothing to validate that all tests have a tag.)

I was also a bit skeptical of the `@nospec` tag when it was added in JS, becuase it meant that clearly (at least to me) had nothing to do with the spec (such as the React Hooks) still ended up being marked with `@nospec`. And in the case of Swift Chat, where we're trying to add unit tests for individual components, that would also extend to all of the test cases for internal components or helpers.

Personally, I think it's a bunch of extra comments for little benefit.

Reply:
> Also what about embedding spec id into the test's name directly? test_CHA_RL1a1_1 as an example.

I mean, we _could_ (but you'd also have to add the information provided by the tag name, i.e. is this a `@spec` or a `@specOneOf(m/n)` or a `@specPartial`?), and it feels like it would be less readable.

Reply:
Also, if you perhaps wanted to test multiple spec points using parameterised tests (which Swift Testing supports), then you’d have to have a _really_ long test name!